code/controllers/subsystem/mapping.dm

```diff
@@ -116,56 +127,41 @@ var/datum/subsystem/mapping/SSmapping
 						CHECK_TICK
 				world.log << "Z-level [z_level] for [P.name] unloaded"
 				deallocate_zlevel(P)
-		for(var/datum/sub_turf_block/STB in split_block(locate(1, 1, 1), locate(255, 255, 1)))
-			for(var/turf/T in STB.return_list())
-				for(var/A in T.contents)
-					qdel(A) // Clear everything out, not including docking ports
-				for(var/A in T.contents)
-					qdel(A) // Some qdels dump their shit on the ground.
-				CHECK_TICK
 	world.log << "Loading z-levels for new sector..."
 	var/list/ruins_levels = list()
 
-	for(var/datum/planet/P in star.planets)
-		for(var/I in 1 to P.map_names.len)
-			var/map_name = P.map_names[I]
-			if(!allocate_zlevel(P, I))
-				world.log << "Skipping [P.z_levels[I]] for [P.name]"
-				continue
-			var/map = "[P.map_prefix][map_name]"
-			var/file = file(map)
-			if(isfile(file))
-				mineral_spawn_override = P.rings_composition
-				maploader.load_map(file, 1, 1, P.z_levels[I])
-
-				smooth_zlevel(P.z_levels[I])
-				world.log << "Z-level [P.z_levels[I]] for [P.name] loaded: [map]"
-			else
-				world.log << "Unable to load z-level [P.z_levels[I]] for [P.name]! File: [map]"
-			if(P.spawn_ruins)
-				ruins_levels += P.z_levels[I]
-			CHECK_TICK
-		
-		P.docks = list()
+	for(var/I in 1 to PL.map_names.len)
+		var/map_name = PL.map_names[I]
+		if(!allocate_zlevel(PL, I))
+			world.log << "Skipping [PL.z_levels[I]] for [PL.name]"
+			continue
+		var/map = "[PL.map_prefix][map_name]"
+		var/file = file(map)
+		if(isfile(file))
+			mineral_spawn_override = PL.rings_composition
+			maploader.load_map(file, 1, 1, PL.z_levels[I])
+
+			smooth_zlevel(PL.z_levels[I])
+			world.log << "Z-level [PL.z_levels[I]] for [PL.name] loaded: [map]"
+		else
+			world.log << "Unable to load z-level [PL.z_levels[I]] for [PL.name]! File: [map]"
+		if(PL.spawn_ruins)
```

Comment:
I'm not sure if we should load ruins. They still don't add anything to gameplay value, they just lag the loader.